3.1.0 changes

cs-abhishek-shukla opened the pull request:
- Added release notes
    - Removed PBs marked deprecated from contents
    - Added filtering records section
    - Added information about voice support in setup, usage, and readme

cs-abhishek-shukla committed:
Merge pull request #1 from fortinet-fortisoar/release/3.1.0

Release/3.1.0

cs-abhishek-shukla committed:
3.1.0 changes
- Added release notes
- Removed PBs marked deprecated from contents
- Added filtering records section
- Added information about voice support in setup, usage, and readme

cs-ankit-gupta commented on docs/setup.md:
```diff
@@ -62,19 +62,29 @@ The following permission allow a user to interact with the **FortiSOAR AI Assist
+>The AI Assistant Utils connector does not require a configuration. For more details, see the [AI Assistant Utils Connector](https://docs.fortinet.com/fortisoar/connectors/ai-assitant-utils) document.
 
-- The **AI Assistant** widget to add the **FortiSOAR AI Assistant** bot to the FortiSOAR user interface. For more details, see the [AI Assistant Widget](https://docs.fortinet.com/document/fortisoar/1.0.0/ai-assistant-utils/690/ai-assistant-utils-v1-0-0) document.
+- The **AI Assistant** widget to add the **FortiSOAR AI Assistant** bot to the FortiSOAR user interface. For more details, see the [AI Assistant Widget](https://docs.fortinet.com/fortisoar/connectors/ai-assitant-utils) document.
```
To AI Assistant widget doc link

cs-suyog-jinturkar replied:
@cs-abhishek-shukla

cs-abhishek-shukla replied:
Pointing to Dev repo as the branch is merged in develop after release, ok? @cs-suyog-jinturkar @cs-ankit-gupta

cs-abhishek-shukla committed:
Note added in release notes regarding mic button unavailability

cs-ankit-gupta commented on release_notes.md:
```diff
@@ -1,59 +1,10 @@
 # What's New
 
-## Support for FortiAI GenAI Assistant
+## Support for Voice Recognition in AI Assistant
```
New release note as that now we are not deleting the thread from OpenAI, we are only clearing the conversation on clicking the 'Clear Conversation' from bot

cs-suyog-jinturkar replied:
@cs-abhishek-shukla

cs-abhishek-shukla replied:
same thread? not new thread? @cs-ankit-gupta

cs-abhishek-shukla committed:
WIP filtering records prompt

cs-suyog-jinturkar commented on docs/prompts-for-filtering-records.md:
LGTM. Please add remaining ones

cs-suyog-jinturkar commented on docs/advanced-usage.md:
link for filtering records prompts needs to be added here in its respective section.

cs-abhishek-shukla committed:
WIP filtering records prompt
- broke the doc in 3 parts
- added link in readme

cs-abhishek-shukla committed:
filtering records prompt
- broke the doc in 3 parts
- added link in readme
- Completed filtering indicator records

cs-rahul-vishwakarma commented on docs/prompts-for-filtering-alert-records.md:
```diff
@@ -0,0 +1,103 @@
+
+1. Give me alerts assigned to <user-1>.
+ 
+2. Give me all alerts generated between October 1, 2024, and October 15, 2024
```
Please change to 
`Give me all alerts created between October 1, 2024, and October 15, 2024`

cs-abhishek-shukla replied:
Done

cs-rahul-vishwakarma commented on docs/prompts-for-filtering-alert-records.md:
```diff
@@ -0,0 +1,103 @@
+3. Show me all alerts where the *severity* is either high or critical, the status is open or in progress, and the type is malware or phishing. The alerts should have been created on within the last 7 days, and the source IP should start with '192.168.1.
+ 
+4. List all in-progress alerts, excluding those of the phishing or suspicious type.
+
```
Add more prompts:
`filter out all the alerts with technique ID T1074.001`
`Find alerts triggered by the rule PH_RULE_UEBA_AI_FILE_WRITTEN`

cs-abhishek-shukla replied:
Done

cs-rahul-vishwakarma commented on docs/prompts-for-filtering-incident-records.md:
```diff
@@ -0,0 +1,39 @@
+
+### Other examples
+
+Similarly, you can use the following prompts to filter incident records.
```
Replace the prompts with following examples:
`Give me incidents led by me.`
`Give me incidents created in last 7 days.`
`Give me incidents tagged as super, and not led by <user-1>`
`fetch incidents of type phishing which are in containment phase`

cs-abhishek-shukla replied:
Done

cs-rahul-vishwakarma commented on docs/prompts-for-filtering-indicator-records.md:
```diff
@@ -0,0 +1,37 @@
+
+### Review Results
+
+Verify that the results include only incidents with an ID greater than 400, ensuring no incidents with a lower ID are included.
```
Please update this review result to followiing:
`Verify that the results include only indicators of type URL are filtered`

cs-abhishek-shukla replied:
Done

cs-rahul-vishwakarma commented on docs/res/filter-alert-assigned-self.png:
Sir, we need to change all the images, as applied filters are not visible. (It's a known issue in 7.6.1)

cs-rahul-vishwakarma commented on release_notes.md:
```diff
@@ -1,59 +1,16 @@
->[!Note]
->Delete these playbooks manually when upgrading from an older **FortiAI** version.
\ No newline at end of file
+Now, threads are retained within OpenAI, enabling improved historical reference and continuity. Each new prompt still initiates a fresh thread, while the user interface continues to clear conversation history after each prompt, ensuring a clean workspace.
```
This sounds confusing! Need to rephrase this.

cs-abhishek-shukla replied:
Changed

cs-rahul-vishwakarma commented on README.md:
```diff
@@ -15,6 +15,16 @@ FortiAI is a context-aware GenAI security assistant that transforms how security
 
+## Voice Support Overview
+
+FortiAI now includes voice-to-natural-language functionality, enabling hands-free interaction. This feature allows users to give voice commands for tasks such as generating playbooks, filtering records, or blocking indicators, streamlining workflows and increasing accessibility. This feature has multiple benefits like the following:
```
It's a **Tap-and-Hold** feature so saying "hands-free interaction" seems counterintuitive.

cs-suyog-jinturkar replied:
let it be.. :)

cs-abhishek-shukla committed:
Added errors and workarounds in troubleshooting

cs-abhishek-shukla committed:
PR comments addressed

cs-abhishek-shukla committed:
PR comments addressed
